server/src/main/java/com/cloud/network/router/VpcNetworkHelperImpl.java

```diff
@@ -87,7 +87,7 @@ public void reallocateRouterNetworks(final RouterDeploymentDefinition vpcRouterD
         final TreeSet<String> publicVlans = new TreeSet<String>();
         if (vpcRouterDeploymentDefinition.isPublicNetwork()) {
             publicVlans.add(vpcRouterDeploymentDefinition.getSourceNatIP()
-                                                         .getVlanTag());
+                                                         .getVlanTag()+"");
```

Comment:
This can be reverted

Reply:
I put this, coz in the case of nsx we get an NPE